File: compilador/src/main/java/org/fipp/compilador/Token.java
package org.fipp.compilador;

public class Token {
    private String lexema;
    private String tipo;
    private int linha;

    public Token(String lexema, String tipo, int linha) {
        this.lexema = lexema;
        this.tipo = tipo;
        this.linha = linha;
    }

    public String getLexema() { return lexema; }
    public String getTipo() { return tipo; }
    public int getLinha() { return linha; }

    @Override
    public String toString() {
        return String.format("Linha %d: <%s, %s>", linha, tipo, lexema);
    }
}

File: src/utils/Token.js
class Token {
    constructor(lexema, tipo, linha) {
      this.lexema = lexema;
      this.tipo = tipo;
      this.linha = linha;
    }
  
    toString() {
      return `${this.tipo}: ${this.lexema} (linha ${this.linha})`;
    }
  }
  
  export default Token;